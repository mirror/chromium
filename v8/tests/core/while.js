print("while (false)");
while (false)
  print("should never reach here");

print("while (true)");
var i = 10;
while (true) {
  print(i);
  i--;
  if (i == 0)
    break;
}

print("while w/ continue");
i = 10;
while (i-- > 0)
  continue;

print("while w/ break");
while (true) {
  print("iteration");
  break;
}

print("while while w/ break");
foo: while (true) {
  print("iteration A");
  bar: while (true) {
    print("iteration B");
    break foo;
  }
}

print("while w/ continue & break");
i = 0;
loop: while (i < 10) {
  i++;
  switch (i) {
    case 5: continue;
    case 8: break loop;
  }
  print(i);
}
