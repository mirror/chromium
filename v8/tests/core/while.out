while (false)
while (true)
10
9
8
7
6
5
4
3
2
1
while w/ continue
while w/ break
iteration
while while w/ break
iteration A
iteration B
while w/ continue & break
1
2
3
4
6
7
